pkg/scheduler/framework/plugins/dynamicresources/extendeddynamicresources.go

```diff
@@ -391,3 +403,164 @@ func createRequestMappings(claim *resourceapi.ResourceClaim, pod *v1.Pod) []v1.C
 	}
 	return cer
 }
+
+// unreserveExtendedResourceClaim cleans up the scheduler-owned extended resource claim
+// when scheduling fails. It reverts the assume cache, and deletes the claim from the API
+// server if it was already created.
+func (pl *DynamicResources) unreserveExtendedResourceClaim(ctx context.Context, logger klog.Logger, pod *v1.Pod, state *stateData) {
+	extendedResourceClaim := state.claims.extendedResourceClaim()
+	if extendedResourceClaim == nil {
+		// there is no extended resource claim
+		return
+	}
+
+	// If the claim was marked as pending allocation (in-flight), remove that marker and restore
+	// the assumed claim state to what it was before this scheduling attempt.
+	if deleted := pl.draManager.ResourceClaims().RemoveClaimPendingAllocation(state.claims.getInitialExtendedResourceClaimUID()); deleted {
+		pl.draManager.ResourceClaims().AssumedClaimRestore(extendedResourceClaim.Namespace, extendedResourceClaim.Name)
+	}
+	if isSpecialClaimName(extendedResourceClaim.Name) {
+		// In memory temporary extended resource claim does not need to be deleted
+		return
+	}
+	// Claim was written to API server, need to delete it to prevent orphaned resources.
+	logger.V(5).Info("delete extended resource backed by DRA", "resourceclaim", klog.KObj(extendedResourceClaim), "pod", klog.KObj(pod), "claim.UID", extendedResourceClaim.UID)
+	extendedResourceClaim = extendedResourceClaim.DeepCopy()
+	if err := pl.deleteClaim(ctx, extendedResourceClaim, logger); err != nil {
+		logger.Error(err, "delete", "resourceclaim", klog.KObj(extendedResourceClaim))
+	}
+}
+
+// deleteClaim deletes the claim after removing the finalizer from the claim, if there is any.
+func (pl *DynamicResources) deleteClaim(ctx context.Context, claim *resourceapi.ResourceClaim, logger klog.Logger) error {
```

Comment:
ditto, is it just copied over? it does not show diff

Reply:
new function